Blofin implementation

luckydss opened the pull request:
Exchange Name
BloFin

URL of the Exchange
https://blofin.com/

URL of the Exchange's API docs
https://blofin.com/docs

Why should this exchange be implemented?
Blofin is a rapidly growing exchange that provides users with fast and reliable trading as well as secure asset management.

pcriadoperez reviewed (commented):
Thanks a lot for contributing @luckydss .

However please review the contributing docs. 
Most of the code we write in typescript following certain rules, and then it's automatically transpiled to other languages. So we look to only commit the .ts files

https://docs.ccxt.com/#/CONTRIBUTING

luckydss commented:
> Thanks a lot for contributing @luckydss .
> 
> However please review the contributing docs. Most of the code we write in typescript following certain rules, and then it's automatically transpiled to other languages. So we look to only commit the .ts files
> 
> https://docs.ccxt.com/#/CONTRIBUTING

Thank you for your reply. Your suggestion is received. We are developing a typescript version and expect to submit it this week.

luckydss commented:
> Thanks a lot for contributing @luckydss .
> 
> However please review the contributing docs. Most of the code we write in typescript following certain rules, and then it's automatically transpiled to other languages. So we look to only commit the .ts files

Hello, thank you again for your suggestions. Then according to your suggestions, we made modifications and added typescript code. Please check again and look forward to your reply.

> > Thanks a lot for contributing @luckydss .
> > However please review the contributing docs. Most of the code we write in typescript following certain rules, and then it's automatically transpiled to other languages. So we look to only commit the .ts files
> > https://docs.ccxt.com/#/CONTRIBUTING
> 
> Thank you for your reply. Your suggestion is received. We are developing a typescript version and expect to submit it this week.

Hello, thank you again for your suggestions. According to your suggestions, we have made modifications and added typescript code. Please check again and look forward to your reply.

carlosmiei commented:
@luckydss thanks, we will review it soon

carlosmiei commented on ts/src/blofin.ts:
```diff
@@ -0,0 +1,2165 @@
+    }
+
+    safeMarket (marketId = undefined, market = undefined, delimiter = undefined, marketType = undefined) {
+        const isOption = (marketId !== undefined) && ((marketId.indexOf ('-C') > -1) || (marketId.indexOf ('-P') > -1));
```
does blofin support options? it does not seem so,  i think we can remove this method

luckydss replied:
Thank you very much for your review, and thank you for the code modifications you made. Yes, your suggestions were absolutely correct, and we have made the necessary changes and updates. We look forward to your review once again.

carlosmiei replied:
@luckydss no problem, there are still some fixes pending, but we will handle it, no worries

luckydss replied:
Great, thank you for letting me know. If there is anything I can do to help, please don't hesitate to reach out. Have a great day!

carlosmiei replied:
@luckydss Sure! You too!

carlosmiei commented:
![blofin-resized](https://github.com/ccxt/ccxt/assets/43336371/99d5b50e-89b7-4d4f-8a61-94377b580d3e)

carlosmiei commented:
![blofin-white-resized](https://github.com/ccxt/ccxt/assets/43336371/e545c50e-d1ef-42f8-becd-bb7bc92ec260)

carlosmiei commented:
![blofin-new-white-resized](https://github.com/ccxt/ccxt/assets/43336371/255a7b29-341f-4d20-8342-fbfae4932807)
